File: Neetcode/SubSets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        output = []
        res = []
        def bt(i):
            if i >= len(nums):
                output.append(res.copy())
                return
            res.append(nums[i])
            bt(i + 1)
            res.pop()
            bt(i + 1)
        bt(0)
        return output

File: Neetcode/SubSets.cpp
class Solution {
public:
    vector<vector<int>> out;
    vector<int> res;
    void bt(vector<int> nums, int i){
        if (i >= nums.size()){
            out.push_back(res);
            return;
        }
        res.push_back(nums[i]);
        bt(nums, i+1);
        res.pop_back();
        bt(nums, i+1);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        bt(nums, 0);
        return out;
    }
};
